scripts/check-RN-link-perms.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+# Checks if a release notes file is updated. If true, searches the release notes file for links to bugs that are behind a login and fails the build.
+# We should not include internal bugs in our release notes.
+
+# Get the repo path and files changed
+REPO_PATH="$(git rev-parse --show-toplevel)"
+FILES=$(git diff --name-only HEAD~1 HEAD --diff-filter=d "*.adoc" ':(exclude)_unused_topics/*')
+
+# Function to check links in updated release notes
+check_rn_links () {
+
+    # Array for release files in case more than one are edited in a PR.
+    release_files=()
+
+    #Extract the release notes file from all the updated files
+    while IFS= read -r line; do
+        if echo "$line" | grep -q "release-notes"; then
+            release_files+=("$line")
+        fi
+    done <<< "$FILES"
+
+    for RELEASE_FILE in "${release_files[@]}"; do
```

Comment:
Is it possible that there is more than one release-notes file per release?

Reply:
I'm not sure but I was trying to handle edge cases. There are a lot of "release-notes" modules in the repo and I don't know about their workflow. So in case they do update OCP RNs and their own sub-release notes... I don't know if this is a thing but conscious that I don't want builds failing and writers unable to get around it.

![image](https://github.com/openshift/openshift-docs/assets/104497497/7e20268e-dbb7-4cf2-9a80-9dd6c5fe95b9)

Reply:
Ah yes, very good.